src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenIdConnectAuthenticationHandler.cs

```diff
@@ -92,13 +110,43 @@ public OpenIdConnectAuthenticationHandler(HttpClient backchannel)
                     message = redirectToIdentityProviderNotification.ProtocolMessage;
                 }
 
-                var redirectUri = message.CreateLogoutRequestUrl();
-                if (!Uri.IsWellFormedUriString(redirectUri, UriKind.Absolute))
+                if (Options.AuthenticationMethod == OpenIdConnectAuthenticationMethod.RedirectGet)
                 {
-                    Logger.LogWarning(Resources.OIDCH_0051_RedirectUriLogoutIsNotWellFormed, redirectUri);
+                    var redirectUri = message.CreateLogoutRequestUrl();
+                    if (!Uri.IsWellFormedUriString(redirectUri, UriKind.Absolute))
+                    {
+                        Logger.LogWarning(Resources.OIDCH_0051_RedirectUriLogoutIsNotWellFormed, redirectUri);
+                    }
+
+                    Response.Redirect(redirectUri);
                 }
+                else if (Options.AuthenticationMethod == OpenIdConnectAuthenticationMethod.FormPost)
+                {
+                    var inputs = new StringBuilder();
+                    foreach (var parameter in message.Parameters)
+                    {
+                        var name = Options.HtmlEncoder.HtmlEncode(parameter.Key);
+                        var value = Options.HtmlEncoder.HtmlEncode(parameter.Value);
 
-                Response.Redirect(redirectUri);
+                        var input = string.Format(CultureInfo.InvariantCulture, InputTagFormat, name, value);
+                        inputs.AppendLine(input);
+                    }
+
+                    var issuer = Options.HtmlEncoder.HtmlEncode(message.IssuerAddress);
+
+                    var content = string.Format(CultureInfo.InvariantCulture, HtmlFormFormat, issuer, inputs);
```

Comment:
Then we'll have to de-constant-ize (I love that verb :smile:) `HtmlFormFormat`.

See this comment for why I prefer avoiding using string interpolation on .NET 4.5.1 (though it's a great feature): https://github.com/aspnet/Security/pull/392#discussion_r36569188